src/integer_mod_q/mat_zq/sample/discrete_gauss.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright © 2023 Niklas Siemer
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module contains algorithms for sampling according to the discrete Gaussian distribution.
+
+use crate::{
+    error::MathError,
+    integer::Z,
+    integer_mod_q::MatZq,
+    rational::Q,
+    traits::{GetNumColumns, GetNumRows, SetEntry},
+};
+use std::fmt::Display;
+
+impl MatZq {
+    /// Initializes a new matrix with dimensions `num_rows` x `num_columns` and with each entry
+    /// sampled independently according to the discrete Gaussian distribution,
+    /// using [`Z::sample_discrete_gauss`].
+    ///
+    /// Parameters:
+    /// - `num_rows`: specifies the number of rows the new matrix should have
+    /// - `num_cols`: specifies the number of columns the new matrix should have
+    /// - `n`: specifies the range from which [`Z::sample_discrete_gauss`] samples
+    /// - `center`: specifies the positions of the center with peak probability
+    /// - `s`: specifies the Gaussian parameter, which is proportional
+    /// to the standard deviation `sigma * sqrt(2 * pi) = s`
+    ///
+    /// Returns a matrix with each entry sampled independently from the
+    /// specified discrete Gaussian distribution.
+    ///
+    /// # Example
+    /// ```
+    /// use qfall_math::integer_mod_q::MatZq;
+    ///
+    /// let sample = MatZq::sample_discrete_gauss(3, 1, &83, &1024, &0, &1.25f32).unwrap();
```

Comment:
```suggestion
    /// let sample = MatZq::sample_discrete_gauss(3, 1, 83, 1024, 0, 1.25).unwrap();
```
This should be possible in combination with the other suggestions below.